clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -510,6 +528,13 @@ raw2trace_t::merge_and_process_thread_files()
                 FATAL_ERROR("Failed to write to output file");
         }
     } while (file_count > 0);
+    // If we used threading pool during profiling, the trace for a thread may be
+    // seen after the thread exit entry, so we can only check whether the unique
```

Comment:
What?  But you can't put a thread exit entry in the final trace here and then put a memref entry for that thread *afterward*.  Is that what's happening due to the timestamp changes?  We have to enforce that order, don't we?